### DISCUSSION TITLE
Simulation takes too long

### DISCUSSION THREAD

**Discussion by Thomason222** [omnetpp/omnetpp#1143] [Q&A]
Hey everyone,
I am trying to run my simulation with high number of nodes. For example when I ran my simulation with 10 000 nodes with 5 repetition, it took me 40 minutes.

Can I somehow increase the speed so the simulation does not take that much time?
I set these parameters in my omnetpp.ini file + I am using the command-line, not the GUI:
   record-eventlog = false
   cmdenv-performance-display = false
   cmdenv-express-mode = true

**Comment by rhornig** [comment]
1. make sure you are using the `release` build of your simulation (i.e. use inet and not inet_dbg)
2. If you have enough RAM, you can try the `opp_runall` and ply with the `-j` option to run multiple runs concurrently. Depending on the RAM size and CPU number, that could give a bit of speed bust for the whol ebatch.
3. Obviously, use a faster machine
4. If it's your own code, profile your code to see if it has some CPU bottleneck
